tmux/list_test.go

```diff
@@ -0,0 +1,98 @@
+package tmux
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestFormat(t *testing.T) {
+	want := "#{session_activity} #{session_alerts} #{session_attached}" +
+		" #{session_attached_list} #{session_created} #{session_format} " +
+		"#{session_group} #{session_group_attached} " +
+		"#{session_group_attached_list} #{session_group_list} " +
+		"#{session_group_many_attached} #{session_group_size} " +
+		"#{session_grouped} #{session_id} #{session_last_attached} " +
+		"#{session_many_attached} #{session_marked} #{session_name} " +
+		"#{session_path} #{session_stack} #{session_windows}"
+	got := format()
+	require.Equal(t, want, got)
+}
+
+// BenchmarkFormat
+//
+//   - Initial:
+//     BenchmarkFormat-10  649407  1830 ns/op  9936 B/op  41 allocs/op
+//
+//   - Refactored:
+//     BenchmarkFormat-10  6360228  181.9 ns/op  448 B/op  1 allocs/op
+func BenchmarkFormat(i *testing.B) {
+	for n := 0; n < i.N; n++ {
+		format()
+	}
+}
+
+func TestProcessSessions(t *testing.T) {
+	testCases := map[string]struct {
+		Input    []string
+		Expected []*TmuxSession
+	}{
+		"Single active session": {
+			Input: []string{
+				"1705879337  1 /dev/ttys000 1705878987 1       0 $2 1705879328 0 0 session-1 /some/test/path 1 1",
+			},
+			Expected: []*TmuxSession{},
+		},
+		"Single inactive session": {
+			Input: []string{
+				"1705879002  0  1705878987 1       0 $2 1705878987 0 0 session-1 /some/test/path 1 1",
+			},
+			Expected: make([]*TmuxSession, 1),
+		},
+		"Two inactive session": {
+			Input: []string{
+				"1705879002  0  1705878987 1       0 $2 1705878987 0 0 session-1 /some/test/path 1 1",
+				"1705879063  0  1705879002 1       0 $3 1705879002 0 0 session-2 /some/other/test/path 1 1",
+			},
+			Expected: make([]*TmuxSession, 2),
+		},
+		"Two active session": {
+			Input: []string{
+				"1705879337  1 /dev/ttys000 1705878987 1       0 $2 1705879328 0 0 session-1 /some/test/path 1 1",
+				"1705879337  1 /dev/ttys000 1705878987 1       0 $2 1705879328 0 0 session-1 /some/test/path 1 1",
+			},
+			Expected: []*TmuxSession{},
+		},
+		"No sessions": {
+			Expected: []*TmuxSession{},
+		},
+		"Invalid LastAttached (Issue 34)": {
+			Input: []string{
+				"1705879002  0  1705878987 1       0 $2 1705878987 0 0 session-1 /some/test/path 1 1",
+				"1705879063  0  1705879002 1       0 $3  0 0 session-2 /some/other/test/path 1 1",
+			},
+			Expected: make([]*TmuxSession, 2),
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			got := processSessions(tc.Input)
+			require.Equal(t, len(tc.Expected), len(got))
+		})
+	}
+}
+
+// BenchmarkProcessSessions
+//   - Initial:
+//     BenchmarkProcessSessions-10  817090  1405 ns/op 1712 B/op  29 allocs/op
+//   - Refactored:
+//     BenchmarkProcessSessions-10  1692590  709.8 ns/op  744 B/op  4 allocs/op
+func BenchmarkProcessSessions(b *testing.B) {
```

Comment:
How does benchmarking work? Does the code comment update or did you just copy/paste the verbose test results?

Furthermore, can we make an assertion to ensure the benchmarking doesn't hit a upper-limit to investigate if tmux has some performance issues in a future version?

Reply:
The benchmark tests get run when by `go test` when you add the appropriate flags. In this case by running `go test -bench=. -benchmem` in the `tmux` directory. 

The comments with the results are something I added manually as I was working through the code.

There isn't anything in the standard lib that would establish a performance criteria for these tests. You could probably implement something manually, but I imagine it would be too flakey to be beneficial. Subtle differences between systems or their running state can have significant effects on the benchmark tests.

Reply:
Do you think it's a good idea to remove the benchmark comments then? I'm not sure what value they have now that you've resolved the performance issue.

Reply:
Sure. Dropped the comments.